include/userobjects/BooleanComboClusteringUserObject.h

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include "GeneralUserObject.h"
+#include <unordered_map>
+
+class ClusteringHeuristicUserObjectBase;
+
+/**
+ * A clustering user object which implements the mesh walking process
+ * and implements the clustering by evaluating  a user defined
+ * boolean expression involving other heuristic based user objects
+ */
+class BooleanComboClusteringUserObject : public GeneralUserObject
+{
+public:
+  static InputParameters validParams();
+  BooleanComboClusteringUserObject(const InputParameters & parameters);
+
+  virtual void execute() override;
+  virtual void initialize() override {};
+  virtual void finalize() override {};
+
+private:
+  /**
+   * getter function for extra element integer score of an element
+   * @param[int] a libmesh element
+   * @param[out] extra element integer score of that element
```

Comment:
Should use `@return` for return values. `@param[out]` is when something is returned by reference through the parameter list of the function (not the case here)